File: Startupba/UI/startupba_desktop/lib/model/support_ticket.dart
import 'package:json_annotation/json_annotation.dart';

part 'support_ticket.g.dart';

@JsonSerializable()
class SupportTicket {
  final int id;
  final int userId;
  final String userName;
  final String subject;
  final String message;
  final int status;
  final String statusName;
  final String? adminResponse;
  final DateTime createdAt;
  final DateTime? answeredAt;
  final DateTime? closedAt;

  SupportTicket({
    this.id = 0,
    this.userId = 0,
    this.userName = '',
    this.subject = '',
    this.message = '',
    this.status = 0,
    this.statusName = '',
    this.adminResponse,
    required this.createdAt,
    this.answeredAt,
    this.closedAt,
  });

  factory SupportTicket.fromJson(Map<String, dynamic> json) =>
      _$SupportTicketFromJson(json);
  Map<String, dynamic> toJson() => _$SupportTicketToJson(this);
}

File: Startupba/Startupba.Services/Database/SupportTicket.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Startupba.Services.Database
{
    /// <summary>
    /// A user support request (complaint, question or suggestion)
    /// handled by the platform administrator.
    /// </summary>
    public class SupportTicket
    {
        [Key]
        public int Id { get; set; }

        [Required]
        public int UserId { get; set; }

        [Required]
        [MaxLength(200)]
        public string Subject { get; set; } = string.Empty;

        [Required]
        [MaxLength(2000)]
        public string Message { get; set; } = string.Empty;

        /// <summary>
        /// 0=Open, 1=Answered, 2=Closed
        /// </summary>
        [Required]
        public int Status { get; set; } = 0;

        [MaxLength(2000)]
        public string? AdminResponse { get; set; }

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        public DateTime? AnsweredAt { get; set; }

        public DateTime? ClosedAt { get; set; }

        // Navigation property
        [ForeignKey("UserId")]
        public User User { get; set; } = null!;
    }
}
